.github/workflows/deploy-happy-stack.yml

```diff
@@ -162,6 +162,140 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+  run-e2e-tests:
+    timeout-minutes: 60
+    name: e2e-tests-${{ matrix.project }}-${{ matrix.shardCurrent }}-${{ matrix.shardTotal }}
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        # Only run Chrome for now, since GHA only has 250 workers and will cancel jobs if it runs out
+        project: [chromium]
+        shardCurrent:
+          [
+            1,
+            2,
+            3,
+            4,
+            5,
+            6,
+            7,
+            8,
+            9,
+            10,
+            11,
+            12,
+            13,
+            14,
+            15,
+            16,
+            17,
+            18,
+            19,
+            20,
+          ]
+        shardTotal: [20]
+
+    defaults:
+      run:
+        working-directory: frontend
+    steps:
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "16.14.2"
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: us-west-2
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          role-duration-seconds: 1800
+      - name: Login to ECR
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ secrets.ECR_REPO }}
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        run: |
+          npm ci
+          npx playwright install --with-deps
+          cp src/configs/local.js src/configs/configs.js
+
+      # Run E2E tests
+      - name: Run E2E tests
+        env:
+          CI: true
+          PROJECT: ${{ matrix.project }}
+          SHARD: ${{ matrix.shardCurrent }}/${{ matrix.shardTotal }}
+          DEPLOYMENT_STAGE: ${{ github.event.deployment.environment }}
+        run: DEBUG=pw:api npm run e2e-${DEPLOYMENT_STAGE}
```

Comment:
this runs the corresponding `npm` command, which uses env variables above to know which shard and project to use